source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,317 @@
+       | ``tlsCertificateKeyFile``, type: ``string``
+       | ``cert``, type: ``Buffer``
+     - Specifies the client certificate file or the client
+       private key file.
+
+   * - ``key``, type: ``Buffer``
+     - | ``tlsCertificateKeyFile``, type: ``string``
+       | ``key``, type: ``Buffer``
+     - Specifies the client private key file.
+
+   * - ``passphrase``, type: ``string``
+     - ``tlsCertificateKeyFilePassword``, type: ``string``
+     - Specifies the password to decrypt the client private key.
+
+.. note:: Additional TLS Client Options
+   
+   For **testing purposes**, you can set the ``tlsAllowInvalidHostnames`` and
+   ``tlsAllowInvalidCertificates`` client options.
+   
+   Setting the ``tlsAllowInvalidHostnames`` option to ``true`` disables
+   hostname verification.
+   Setting the ``tlsAllowInvalidCertificates`` to ``true`` disables
+   certificate validation and allows you to test a connection with an
+   invalid certificate, such as a self-signed certificate.
+   
+   Specifying either of these options in a production environment puts
+   your application at risk of becoming
+   `insecure <https://tlseminar.github.io/docs/mostdangerous.pdf>`__ or
+   vulnerable to expired certificates and to foreign processes posing
+   as valid client instances.
+
+For a full list of client options, see :ref:`node-connection-options`.
+
+.. _node-client-tls-connect:
+
+Reference Certificates in a Client
+----------------------------------
+
+You must reference your certificates in your ``MongoClientOptions`` so
+that the Server can validate them before the client connects. The
+driver provides multiple TLS-related client options to reference your
+certificates. You can connect with the following option configurations:
+
+- Passing a ``SecureContext`` object to the ``secureContext`` option *(Recommended)*
+- Passing filepath strings to the ``tlsCAFile`` and
+  ``tlsCertificateKeyFile`` options
+- Passing ``Buffer`` objects that contain the contents of your
+  certificates to the ``ca``, ``cert``, and ``key`` options
+
+.. _node-tls-securecontext:
+
+Create a ``SecureContext`` Object to Connect
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We recommend that you use the ``secureContext`` option to connect with
+the TLS protocol. By using a ``SecureContext`` object to store your
+certificates, you use the Node.js framework for TLS that is independent of the
+driver. When you use this option, you keep all TLS options in a
+reusable object. This is the most organized way to set up
+TLS for your application.
+
+To create a ``SecureContext`` object, you must first install the ``tls``
+package and use it in your application. The ``tls.createSecureContext()``
+method takes an options parameter that contains the contents of the
+certificates. The ``secureContext`` option in your
+``MongoClientOptions`` takes the ``SecureContext`` object as its value,
+allowing the client to reference the certificates for validation.
```

Comment:
I: These couple of sentences would be clearer if they were imperative/second-person

S:

```suggestion
package and use it in your application. Next, call the ``tls.createSecureContext()``
method and pass in the contents of the certificates as an options object. 
This method returns a ``SecureContext`` object that you can include as an option 
in your ``MongoClientOptions`` object.
```

Reply:
I wasn't sure if these sections should be structured as "tutorials," but I think given the action-oriented titles of the subsections, this seems like an appropriate change.